Quick Tags: Work correctly on non-NPF posts

marcustyphoon opened the pull request:
### Description
<!--
  What is the goal of this pull request?
  How does it achieve that goal?
  Any other context needed to understand the changes?

  Please properly link any issues that this PR aims to resolve:
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

This adds an alternative API endpoint call to Quick Tags on posts that we cannot ensure the NPF edit endpoint will work on. It uses the mass post editor endpoint, which should work on both legacy and NPF posts.

For discussion about which posts are NPF-compatible, see [discord.com/channels/104051306309644288/327155782934200320/1074286553733533738](https://discord.com/channels/104051306309644288/327155782934200320/1074286553733533738).

Resolves #940 (most likely reason). Resolves #965. Resolves #1224.

### Testing steps
<!--
  What is the intended behaviour of this pull request?
  How exactly can a maintainer reproduce it?

  Please assume your reviewer will load the addon in a temporary profile.
  Feel free to upload a configuration file if the setup is complex.
-->

marcustyphoon commented:
- [ ] see if the mega editor endpoint changes tipping

AprilSylph commented on src/scripts/quick_tags.js:
```diff
@@ -146,26 +147,33 @@ const togglePostOptionPopupDisplay = async function ({ target, currentTarget })
+
+      notify(displayText);
+    } else {
+      await megaEdit([postId], { mode: 'add', tags: tagsToAdd });
+      notify(`Edited on ${blogName} (legacy)`);
```
will need to change the `catch` logic to account for this, do we even get a JSON response from the mass post editor?

marcustyphoon replied:
See #971.

marcustyphoon commented:
This next commit addresses the above comment and #1046, though I'm not sure if this is the ideal way to do so (vs, for example, making the mega editor error wrapped in `{ body: contentsGoHere }`.

marcustyphoon commented:
Besides the virtual scroller stuff, this is IMO a good priority, as the virtual scroller broke legacy XKit's Quick Tags and that seems like a great ~~excuse to have people switch~~ ahem, reason to ensure we have a good alternative.

AprilSylph commented on src/scripts/quick_tags.js:
```diff
@@ -159,15 +160,20 @@ const addTagsToPost = async function ({ postElement, inputTags = [] }) {
+      notify(displayText);
+    } else {
+      await megaEdit([postId], { mode: 'add', tags: tagsToAdd });
+      notify(`Edited on ${blogName} (legacy)`);
```
could make this copy more accurate

```suggestion
      notify(`Edited legacy post on ${blogName}`);
```

marcustyphoon commented:
Now that I think about it, it's probably better not to mention legacy posts in the copy at all, considering that there is some ambiguity there.

AprilSylph commented:
Eh, it's accurate. If we're displaying that message then it's definitely a legacy post, even if it doesn't display for all legacy types.
